python-checks/src/main/java/org/sonar/python/checks/HardCodedCredentialsCheck.java

```diff
@@ -0,0 +1,286 @@
+      && isCredential(((Name) ((RegularArgument) callExpression.arguments().get(0)).expression()).name(), variablePatterns());
+  }
+
+  private static boolean isFirstArgumentAStringLiteral(CallExpression callExpression) {
+    return callExpression.arguments().get(0).is(Tree.Kind.REGULAR_ARGUMENT)
+      && ((RegularArgument) callExpression.arguments().get(0)).expression().is(Tree.Kind.STRING_LITERAL);
+  }
+
+  private boolean isQualifiedExpressionCredential(QualifiedExpression qualifiedExpression) {
+    return qualifiedExpression.qualifier().is(Tree.Kind.NAME) && isCredential(((Name) qualifiedExpression.qualifier()).name(), variablePatterns());
+  }
+
+  private static void checkSensitiveArgument(CallExpression callExpression, int argNb, SubscriptionContext ctx) {
+    for (int i = 0; i < callExpression.arguments().size(); i++) {
+      Argument argument = callExpression.arguments().get(i);
+      if (argument.is(Tree.Kind.REGULAR_ARGUMENT)) {
+        RegularArgument regularArgument = (RegularArgument) argument;
+        if (regularArgument.keywordArgument() != null) {
+          return;
+        } else if (i == argNb && regularArgument.expression().is(Tree.Kind.STRING_LITERAL)) {
+          ctx.addIssue(callExpression, MESSAGE);
+        }
+      }
+    }
+  }
+
+  private void handleStringLiteral(StringLiteral stringLiteral, SubscriptionContext ctx) {
+    if (stringLiteral.stringElements().stream().anyMatch(StringElement::isInterpolated)) {
+      return;
+    }
+    if (isCredential(stringLiteral.trimmedQuotesValue(), literalPatterns())) {
+      ctx.addIssue(stringLiteral, MESSAGE);
+    }
+    if (isURLWithCredentials(stringLiteral)) {
+      ctx.addIssue(stringLiteral, MESSAGE);
+    }
+  }
+
+  private static boolean isURLWithCredentials(StringLiteral stringLiteral) {
+    try {
+      URL url = new URL(stringLiteral.trimmedQuotesValue());
+      String userInfo = url.getUserInfo();
+      if (userInfo != null && userInfo.matches("\\S+:\\S+")) {
+        return true;
+      }
+    } catch (MalformedURLException e) {
+      return false;
+    }
+    return false;
+  }
+
+  private void handleBinaryExpression(BinaryExpression binaryExpression, SubscriptionContext ctx) {
+    boolean shouldReport = false;
+    if (binaryExpression.leftOperand().is(Tree.Kind.NAME, Tree.Kind.QUALIFIED_EXPR) && binaryExpression.rightOperand().is(Tree.Kind.STRING_LITERAL)) {
+      shouldReport = checkBinaryExpression(binaryExpression.leftOperand());
+    }
+    if (binaryExpression.rightOperand().is(Tree.Kind.NAME, Tree.Kind.QUALIFIED_EXPR) && binaryExpression.leftOperand().is(Tree.Kind.STRING_LITERAL)) {
+      shouldReport = checkBinaryExpression(binaryExpression.rightOperand());
+    }
+    if (shouldReport) {
+      ctx.addIssue(binaryExpression, MESSAGE);
+    }
+  }
+
+  private boolean checkBinaryExpression(Expression operand) {
+    if (operand.is(Tree.Kind.NAME)) {
```

Comment:
what do you think about checking the symbol name instead (and having as parameter a `HasSymbol expression`)?

Reply:
The advantage of the current implementation is we can report issues even if we can't resolve the symbol.
That said I agree the code would look much cleaner using `HasSymbol`, so I don't have a big opinion tbh, do you think I should go for HasSymbol?